### ISSUE TITLE
Converting NRRD file to Dicom file 

### ISSUE THREAD

**Issue by NonlinearNimesh** [mhe/pynrrd#115]
Hi I have a .nrrd file and I want to convert that image into Dicom file, Can anyone help me in doing that

for now i tried doing like NRRD -> GIF -> JPG -> Dicom but the final results was not good, can any one please help me to do the same.

This is the sample file on which i am trying the conversion

https://drive.google.com/file/d/1SII7xVgUhPAH5Bkzg68dF4CAXawsWk9g/view?usp=sharing

Thanks

**Comment by NonlinearNimesh** [comment]
Hi Thanks for the response, 
But in the CT nrrd file there are 504 instances and I needed to convert each and every instance in dicom which I am not able to using the way you mentioned

**Comment by addisonElliott** [comment]
I haven't worked with DICOM in any meaningful way in a few years, so take this with a grain of salt.

I recall that DICOM only supported 2D images. In the research lab I worked in, we had cardiac MRI cine's which were 4 dimensional (3D images of the heart across time). The DICOM files were actually a folder containing many 2D slices. Each 2D slice contains information about where they're located. Any DICOM viewer will load all DICOM files in that folder and peice together the data.

In our lab, we went the other way with the data. The data came in as DICOM and I used `pydicom` to load that data and turn it into NRRD because I felt it was easier to work with.

Long story short, you have to split the data up into 2D slices and save each slice individually in a DICOM file. You'll want to make sure you set the patient, study, and series IDs appropriately.

Welcome to the confusing world of DICOM 😄

**Comment by NonlinearNimesh** [comment]
So like I have NRRD file of size (512,512,504) means There are 504 instances of (512, 512) size image, and what you are trying to say is I pick each instance like (512, 512, 1), (512, 512, 1), (512, 512, 1) till 504 and convert the same into dicom.

**Comment by NonlinearNimesh** [comment]
i am trying to convert NRRD file to dicom using JPG for example NRRD -> JGP -> DICOM
but when i try to convert the NRRD to jpg, output image is noisey
Below is my implementation
```
filename = "/content/drive/MyDrive/CT.nrrd"
readdata, header = nrrd.read(filename)
print(readdata.shape) # (512, 512, 504)

#Not converting pixel to a jpg image 

from PIL import Image
import numpy as np
for i in range(504):
  if i == 200:
    img = np.array(readdata[:,:,i]) # for now i was trying to convert instance number 200 to jpg
    print(img.shape)
    print(np.amax(img))
    #img = (np.maximum(img, 0) / img.max()) * 255.0
    img = np.amax(img) - img
    img = Image.fromarray(np.uint8(img))
    img.save(f'/content/testrgb{i}.png')
    break

But image was unacceptable.

Can you please help in this .................
